### DISCUSSION TITLE
IterativeSplineParameterization::compute(const std::vector<std::reference_wrapper<Instruction>>& trajectory, ...) is confusing

### DISCUSSION THREAD

**Discussion by johnwason** [tesseract-robotics/tesseract#473] [General]
`IterativeSplineParameterization::compute(const std::vector<std::reference_wrapper<Instruction>>& trajectory, ...)` uses `const&` for the `trajectory` parameter, even though this is an in/out parameters. While this may be technically correct, it is confusing since `const&` parameters are typically input only, and SWIG makes that assumption in a lot of places. It would be better to drop the `const` and leave it as a reference.

**Comment by Levi-Armstrong** [comment]
I don't see any issues dropping the const. @mpowelson Do you have any issue with dropping the const?

**Reply by mpowelson** [reply]
Not offhand, but I'd have to try it and see if it causes problems.
